### PR TITLE
Simplify package setup

### PR DESCRIPTION
**Pull request by larrybradley**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires = ["setuptools>30.3.0", "wheel", "numpy>=1.13"]
```
Comment (astrofrog) [inline pyproject.toml]: I wouldn't recommend adding this yet but adding it in #915 since it's best done when removing astropy-helpers

Comment (larrybradley) [reply pyproject.toml]: Will do, thanks!

**Comment by larrybradley** [comment]
@astrofrog, @bsipocz: The python 3.5, numpy 1.13, astropy LTS job is failing with seg faults trying to compile the cython files.  Any idea what's going on there?

https://travis-ci.org/astropy/photutils/jobs/568857136#L3062

**Comment by bsipocz** [comment]
weird, as with a quick glance I don't see any version number differences with the installed dependencies.

**Comment by astrofrog** [comment]
Very strange, I'll try on my Linux desktop shortly.

**Comment by larrybradley** [comment]
@astrofrog Did you find anything?  I'm going to start digging into this.  I'm planning to release 0.7 today (🤞) and I'd like to include this if possible.

**Comment by astrofrog** [comment]
@larrybradley - testing now

**Comment by larrybradley** [comment]
Many thanks!  I haven't started on this yet.

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+packages = find:
+setup_requires =
+    numpy>=1.13
+    cython>=0.28
+install_requires =
+    numpy>=1.13
+    astropy>=2.0
+tests_require =
+    pytest
+    pytest-astropy
```
Comment (astrofrog) [inline setup.cfg]: Ok I thought this bug seemed familiar. The issue is that cython can segfault if psutil is installed. Don't ask... (@bsipocz @pllim - do you know if we had an issue for that somewhere? I know we've discussed it but is there an issue?)

The workaround is to not add pytest-astropy here, but pytest-remotedata and pytest-doctestplus, as it's pytest-openfiles that drags in psutils.

File: setup.cfg [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: Implementation :: CPython
+    Topic :: Scientific/Engineering :: Astronomy
+
+[options]
+python_requires = >=3.5
+packages = find:
+setup_requires =
+    numpy>=1.13
+    cython>=0.28
```
Comment (astrofrog) [inline setup.cfg]: Here I think you should use an approach similar to astropy core, to only use cython if we are in dev mode (not for a release):

https://github.com/astropy/astropy/blob/master/setup.py#L24

File: setup.cfg (hunk 2 above)
Comment (larrybradley) [reply setup.cfg]: Hooray!  Huge thanks!  :tada:

Does `cython` segfault with `psutil` only for python 3.5?

Comment (astrofrog) [reply setup.cfg]: It seems so yes.

File: setup.cfg (hunk 3 above)
Comment (larrybradley) [reply setup.cfg]: OK, will do, thanks!

File: setup.cfg (hunk 2 above)
Comment (bsipocz) [reply setup.cfg]: This vaguely rings a bell, but I don't think we have an issue for it. Is it an issue with a particular psutils version? If yes, maybe we can pin the version in `pytest-astropy` or in `pytest-openfiles`?

Comment (larrybradley) [reply setup.cfg]: Time to drop python 3.5.... 😈

Comment (bsipocz) [reply setup.cfg]: core dropped it, so it's a reasonable move. (hurray f-strings)

Comment (pllim) [reply setup.cfg]: I searched in several repo of interest but can't find the issue either. 🤷‍♀

File: setup.cfg (hunk 3 above)
Comment (larrybradley) [reply setup.cfg]: Done!

File: setup.cfg (hunk 2 above)
Comment (larrybradley) [reply setup.cfg]: Indeed, f-strings are very nice.  I'll keep py3.5 for this release, but drop it for v0.8.

**Comment by larrybradley** [comment]
@astrofrog This is ready for final review.

File: setup.cfg (hunk 3 above)
Comment (larrybradley) [reply setup.cfg]: `is_distutils_display_option` comes from `astropy-helpers`.  Will we need a replacement for this without helpers?

**Comment by larrybradley** [comment]
I've implemented the requested changes and the test pass now.  The `astropy-dev` test failure is unrelated.  Thanks again, @astrofrog.

**Pull request merged**